test/testdata/infer/shape_merge.rb

```diff
@@ -40,3 +40,15 @@ def has_kwargs(a:, b:, c:)
 end
 
 has_kwargs({}.merge(a: 1).merge(b: 3).merge(c: 4))
+
+
+class Key; end
+
+def has_opts_hash(opts={}); end
+
+key = Key.new
+
+# guard against non-symbol values being used as keyword args
+has_opts_hash(**{a: 10}.merge(key => "value"))
```

Comment:
Should there be an error here somewhere? This comment makes me feel like there should. Or is the guard that we don't crash?

Reply:
I should clarify the comment: this is guarding that the case is handled correctly internally, so not crashing is the goal.